mesonpy/_editable.py

```diff
@@ -304,12 +304,26 @@ def rebuild(self) -> Node:
         env[MARKER] = os.pathsep.join((env.get(MARKER, ''), self._build_path))
 
         if self._verbose or bool(env.get(VERBOSE, '')):
-            print('+ ' + ' '.join(self._build_cmd))
-            stdout = None
+            # We do not want any output if there is no work to do. Code is adapted from:
+            # https://github.com/mesonbuild/meson/blob/a35d4d368a21f4b70afa3195da4d6292a649cb4c/mesonbuild/mtest.py#L1635-L1636
+            if self._build_cmd[0] == 'meson':
```

Comment:
The build command on Windows can be something else that the literal string `meson`.

Reply:
Do you have any advice on a more robust way to handle the Windows case?

Reply:
The easiest is probably to check for `if sys.platform == 'win32'`. If that is true, the command being executed is `$MESON compile` possibly with a `--ninja-args=...` argument. Otherwise is `ninja` or something with a compatible command line interface.